src/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs

```diff
@@ -216,26 +211,32 @@ public IntPtr Handle
         [Browsable(false)]
         public int Height => Size.Height;
 
-        public Size Size
+        public unsafe Size Size
         {
             get
             {
                 if (_iconSize.IsEmpty)
                 {
                     var info = new SafeNativeMethods.ICONINFO();
-                    SafeNativeMethods.GetIconInfo(new HandleRef(this, Handle), info);
-                    var bmp = new SafeNativeMethods.BITMAP();
+                    SafeNativeMethods.GetIconInfo(new HandleRef(this, Handle), ref info);
+                    var bitmap = new SafeNativeMethods.BITMAP();
 
                     if (info.hbmColor != IntPtr.Zero)
                     {
-                        SafeNativeMethods.GetObject(new HandleRef(null, info.hbmColor), Marshal.SizeOf(typeof(SafeNativeMethods.BITMAP)), bmp);
+                        SafeNativeMethods.GetObject(
+                            new HandleRef(null, info.hbmColor),
+                            sizeof(SafeNativeMethods.BITMAP),
+                            ref bitmap);
                         SafeNativeMethods.IntDeleteObject(new HandleRef(null, info.hbmColor));
-                        _iconSize = new Size(bmp.bmWidth, bmp.bmHeight);
+                        _iconSize = new Size((int)bitmap.bmWidth, (int)bitmap.bmHeight);
                     }
                     else if (info.hbmMask != IntPtr.Zero)
                     {
-                        SafeNativeMethods.GetObject(new HandleRef(null, info.hbmMask), Marshal.SizeOf(typeof(SafeNativeMethods.BITMAP)), bmp);
-                        _iconSize = new Size(bmp.bmWidth, bmp.bmHeight / 2);
+                        SafeNativeMethods.GetObject(
+                            new HandleRef(null, info.hbmMask),
+                            sizeof(SafeNativeMethods.BITMAP),
+                            ref bitmap);
+                        _iconSize = new Size((int)bitmap.bmWidth, (int)(bitmap.bmHeight / 2));
```

Comment:
Why are the casts necessary now?

Reply:
Because I changed the struct to the native uint. Felt it was better to express that there is no "negative" in the native representations.

Reply:
Ok cool.